core/src/test/java/io/grpc/internal/AbstractTransportTest.java

```diff
@@ -2042,6 +2042,10 @@ public void clientChecksInboundMetadataSize_trailer() throws Exception {
     assertNull(metadata.get(tellTaleKey));
   }
 
+  private boolean shouldServerFlushHeaders() {
+    return !methodDescriptor.getType().serverSendsOneMessage();
```

Comment:
The method type used in these tests is UNKNOWN, so they just do whatever they want to, essentially (UNKNOWN works like BIDI_STREAMING, but the transport generally doesn't care too much). That is to say, this is a bit misleading and tests can use other values, so I think we should just hard-code true/false in the tests of this class. (Feel free to do `/*flush=*/ true` or the like it make it more clear, if you'd like.)

Reply:
Changed to hardcoded flush parameter.